zap/src/main/java/org/parosproxy/paros/network/HttpRequestHeader.java

```diff
@@ -243,8 +244,12 @@ public HttpRequestHeader(String method, URI uri, String version)
 
         setHeader(ACCEPT_ENCODING, null);
 
-        // ZAP: changed from method to version
-        if (version.equalsIgnoreCase(HTTP11)) {
+    // Content-Length is valid only  for POST,PUT, PATCH, DELETE methods
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+    // Per RFC we should not add this header for other methods
+    // https://datatracker.ietf.org/doc/html/rfc7230#section-3.3.2
+    if (method.equalsIgnoreCase(POST) || method.equalsIgnoreCase(PUT) || 
+        method.equalsIgnoreCase(PATCH) || method.equalsIgnoreCase(DELETE)) {
             setContentLength(0);
         }
```

Comment:
I'd suggest just remove it, the components sending bodies already add/update the Content-Length when needed (if not they should).

Reply:
Thanks for the suggestion.. Done!